Source/EasyNetQ/Consumer/HandlerCollection.cs

```diff
@@ -71,13 +67,15 @@ public HandlerCollection(IEasyNetQLogger logger)
             return (message, info) => Task.Factory.StartNew(() => { });
         }
 
-        public dynamic GetHandler(Type messageType)
+        public Func<IMessage<object>, MessageReceivedInfo, Task> GetHandler(Type messageType)
         {
             Preconditions.CheckNotNull(messageType, "messageType");
-
+            
             var getHandlerGenericMethod = GetType().GetMethod("GetHandler", new Type[0]).MakeGenericMethod(messageType);
```

Comment:
It turns out that we need that, otherwise you took an `AmbiguousMatchException` because `GetHandler` is an overload and you must specify you want the empty parameter method, but I found a better way that instantiate a new array, using `Type.EmptyTypes` as the documentation suggest.